### DISCUSSION TITLE
how to make enable/disable global bypass keyboard shortcut (flatpak version)

### DISCUSSION THREAD

**Discussion by rstx32** [wwmm/easyeffects#1945] [Show and tell]
### problem story
i'm a little bit struggling with enabling/disabling easyeffects quickly, because i have two mode listening device:
- speaker
- headphone/iem/earbuds

_i just need easyeffects for my speaker, so the question is : **how can i disable (global bypass) quickly while i switch to headphone/iem/earbuds just using my keyboard?**_

### gathering information
after doing some references research, i got script that showing up global bypass value :
`flatpak run --command=gsettings com.github.wwmm.easyeffects get com.github.wwmm.easyeffects bypass`
that returning boolean value : **true** or **false**

i can enable global bypass by using script :
`flatpak run --command=gsettings com.github.wwmm.easyeffects set com.github.wwmm.easyeffects bypass true`

or disable global bypass by using script : 
`flatpak run --command=gsettings com.github.wwmm.easyeffects set com.github.wwmm.easyeffects bypass false`

### solution
1. with those script i wrote :
![Screenshot_20221107_173720](https://user-images.githubusercontent.com/40822335/200289870-af78c9bc-24de-4b5c-9d18-998c7803ec55.png)
download : [easyeffects-bypass.zip](https://github.com/wwmm/easyeffects/files/9950600/easyeffects-bypass.zip)

2. save to file : `easyeffects-bypass.sh`
3. make it executable : `chmod +x easyeffects-bypass.sh`
4. add custom keyboard shortcut (i'm using KDE), attach `easyeffects-bypass.sh` and configure your own `shortcut trigger`
![Screenshot_20221107_174046](https://user-images.githubusercontent.com/40822335/200291010-a33d5e62-08db-422c-99a3-962bfbb869aa.png)
![Screenshot_20221107_174223](https://user-images.githubusercontent.com/40822335/200291015-1c4923aa-7dce-42fc-aaec-7e82486370d4.png)

that's all! now i can quickly enable or disable easyeffects global bypass just by using keyboard shortcut!

**Comment by wwmm** [comment]
In the PipeWire tab define an autoloading profile for the speaker associated to a preset file that has the effects you want. After plugging you headphones create a preset that has no effects and use it when saving an autoloading profile for the headphones. If everything goes well from this point on the empty preset will be applied whenever your headphones are plugged.

**Reply by rstx32** [reply]
in my case, i'm using external DAC.
- case 1 : using my headphone/iem/earbuds, computer -> DAC -> headphone amplifier -> headphone/iem/earbuds
- case 2 : using my speaker, computer -> DAC -> headphone amplifier as a preamp -> power amplifier -> speakers

so i didn't use 3.5mm motherboard jack at all, the autoloading isn't useful since everything goes to my external dac.

**Reply by wwmm** [reply]
Oh. I see... In this case I think that a script like the one you have already written is the only solution.
